code-challenges.test.js

```diff
@@ -38,11 +71,35 @@ const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 
 // b) Create the function that makes the test pass.
 
+// -->Psuedocode
+// Create a function called newArray that takes an array and filters out the odd numbers using .filter, then sorts the odd numbers using .sort.  Return the sorted odd numbers using .map to ensure they are in order of value 
+
+
+function newArray(array) {
+    const odd = fullArr1.filter((x) => x % 2).sort((a,b) => a - b);
+    return fullArr1.map((x) => x % 2 ? odd.shift() : x);
+ }
+  console.log(newArray([fullArr1]))
```

Comment:
The logic here is so close!  The map function might be adding some empty values to the array here.  Line 79 has almost all the things you need to solve this problem.  Just keep in mind that filter needs to evaluate something and you can append multiple conditions in there using the Logical and (&&) or  or (||).